plugins/source-terasology-modules/gatsby-node.js

```diff
@@ -0,0 +1,169 @@
+    },
+  });
+
+  const lastFetchedKey = "terasology-modules-last-fetched";
+  const dataKey = "terasology-modules-data";
+
+  const now = DateTime.utc();
+  const lastFetched = DateTime.fromISO(await cache.get(lastFetchedKey), {
+    zone: "utc",
+  });
+
+  let repositories = [];
+
+  if (lastFetched.plus({ hours: 12 }) > now) {
+    reporter.info(
+      `[${PLUGIN_NAME}] Loading Terasology module info from cache ...`
+    );
+    repositories = JSON.parse(await cache.get(dataKey));
+  } else {
+    reporter.info(
+      `[${PLUGIN_NAME}] Fetching Terasology module info from GitHub ...`
+    );
+    let hasNextPage = true;
+    let cursor;
+
+    /* Loop iterations depend on the outcome of the previous request, thus 'await' is required here. */
+    /* eslint-disable no-await-in-loop */
+    while (hasNextPage) {
+      const { organization } = await gql(query, { cursor });
+
+      organization.repositories.nodes.forEach((repo) =>
+        repositories.push(repo)
+      );
+
+      hasNextPage = organization.repositories.pageInfo.hasNextPage;
+      cursor = organization.repositories.pageInfo.endCursor;
+    }
+
+    await cache.set(dataKey, JSON.stringify(repositories));
+    await cache.set(lastFetchedKey, now.toISO());
+  }
+
+  reporter.success(`[${PLUGIN_NAME}] Loaded ${repositories.length} modules.`);
+
+  let created = 0;
+  repositories
+    .forEach((repo) => {
+      if (!repo.moduleTxt) {
+        // skip non-module repositories
+        return;
+      }
+
+      let moduleTxt;
+      try {
+        moduleTxt = JSON.parse(repo.moduleTxt?.text);
+      } catch (err) {
+        console.warn(
+          `[${PLUGIN_NAME}] Could not parse 'module.txt' of ${repo.url}.`
+        );
+        return;
+      }
+
+      const tags = [
+        moduleTxt.isTutorial ? "Tutorial" : undefined,
+        moduleTxt.isAsset ? "Asset" : undefined,
+        moduleTxt.isLibrary ? "Library" : undefined,
+        moduleTxt.isSpecial ? "Special" : undefined,
+        moduleTxt.isRendering ? "Rendering" : undefined,
+        moduleTxt.isAugmentation ? "Augmentation" : undefined,
+        moduleTxt.isServerSideOnly ? "Server" : undefined,
+        moduleTxt.isGameplay ? "Gameplay" : undefined,
+      ].filter((x) => x);
+
+      const { id, version, displayName, description, dependencies } = moduleTxt;
+
+      const cover = repo.usesCustomOpenGraphImage
+        ? repo.openGraphImageUrl
+        : undefined;
+
+      const moduleInfo = {
+        id,
+        version,
+        displayName,
+        description,
+        dependencies,
+        tags,
+      };
+
+      moduleInfo.dependencies = moduleInfo.dependencies.map((dep) => ({
+        ...dep,
+        optional: /true/i.test(dep.optional),
+      }));
+
+      const node = {
+        ...repo,
+        usesCustomOpenGraphImage: undefined,
+        openGraphImageUrl: undefined,
+        cover,
+        moduleTxt: moduleInfo,
+        id: createNodeId(`TerasologyModule-${repo.id}`),
```

Comment:
@jdrueckert here we include the whole `repo` object in the node, which includes everything we query via GraphQL (for instance, the `readme`). I explicitly _remove_ the `openGraph...` fields by setting them to `undefined` as the information is encoded in the `cover` field (pointing to the URL of the image if it is a custom image, not set otherwise).

The `moduleTxt` text blob we queried via GraphQL is replaced by the parsed `moduleInfo` (made available under the name `moduleTxt`). 

Maybe these are too many customizations and overrides. I'll consider explicitly assembling the content of the node object in future iterations.

Reply:
:bulb: I see, okay. I was just wondering, why for instance we check whether module.txt is there (obviously because we want to parse it at this point here, yes), but not whether the readme exists...